nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -663,59 +671,141 @@ def is_in_subset(self, peer):
 
         return False
 
+    def compute_subset(self, peers):
+        """
+        Compures all peers that are in the defined subset out of the given peer set
+        :param PeerSet peers: the given peer set
+        :return: peers in the defined subset
+        """
+        if not self.output_config.subset:
+            return peers
+        res = PeerSet()
+        for peer in peers:
+            if self.is_in_subset(peer):
+                res.add(peer)
+        return res
+
     @staticmethod
     def are_labels_all_included(target_labels, pool_labels):
         for key, val in target_labels.items():
             if pool_labels.get(key) != val:
                 return False
         return True
 
-    def exec(self):
+    def exec(self):  # noqa: C901
```

Comment:
simplify this function, consider splitting to multiple functions

Reply:
Done